apps/blaze-dashboard/src/app.jsx

```diff
@@ -73,16 +73,18 @@ async function AppBoot() {
 		window.location.hash = fixPath( window.location.hash );
 	}
 
-	// Ensure locale files are loaded before rendering.
-	setLocale( localeSlug ).then( () => {
-		registerBlazeDashboardPages( window.location.pathname + window.location.search );
+	window.addEventListener( 'load', function () {
```

Comment:
The main problem in the app is that the page finishes loading after the app is fully rendered. For some reason, the page load event is waiting for all of the async calls (including the one we do with hook to the public API) to finally give the page load event.

I don't know what changed in Calypso (or browsers) that caused it to start working that way, but it didn't happen some time ago.

In this video, you can check that the tab page loading continues even after the actual page is rendered and functional. 
![2024-07-04 13 28 04](https://github.com/Automattic/wp-calypso/assets/1258162/8f0cb538-68e8-4c55-b93d-3ce52c31cb87)
If the user tries to change tabs before the tab finish loading, it won't work (the rest of the app will work ok).

It's true that this solution will make the page load a bit slower, but it handles the async call we make in the app a bit better. However, I am not entirely sure about this solution, either; that's why I wanted a second opinion here.

--- 

I just debugged the code a bit more, and the problem appears to be more visual in the link redirection (a href). I noticed that in Jetpack, we have code that adds support for hashbang navigation for those links, and that code runs at page load, so the problem is probably because of that.

It would be a bit hard to find which change affected these apps, but I will try to see if we can improve the Jetpack code to not rely on the page load event. 

I will change the PR to Draft until I find a better solution to the problem. If you by any chance, are aware of a change that could affect how we load the page in Calypso, please let me know 💪. Maybe I need to add something else to the app startup process, or change how we handle our routes.

Reply:
The `load` event is fired only after all scripts, styles, iframes and other resources on the page are finished loading. The `AppBoot` function is running inside such a script, and if it adds another `script` tag to the page, it will delay the `load` event.

I think it's impossible for REST requests, i.e., fetch/XHR requests, to delay the `load` event. Only things that are in the HTML markup can do that.

`script` tags are typically added when webpack is loading a dynamic chunk. If the `setLocale` call or `page.show` ends up loading a webpack chunk, it will delay the `load`.

This patch is kind of making the `load` event "fake", in a sense that when it fires, the page is not really displayed yet. It's only the `page.show` call that does the first React render. Until they, you'll see just some empty placeholder.

To figure out what's really delaying the `load` event, I recommend doing a performance recording of the page load. All the loads are visible in the big waterfall chart, and the `load` event is clearly marked with a vertical line.

Reply:
Thanks for the info! I will do that.